packages/plugin/src/theia.d.ts

```diff
@@ -10263,6 +10263,26 @@ export module '@theia/plugin' {
             options: FormattingOptions,
             token: CancellationToken | undefined
         ): ProviderResult<TextEdit[] | undefined>;
+
+        /**
+         * Provide formatting edits for multiple ranges in a document.
+         *
+         * This function is optional but allows a formatter to perform faster when formatting only modified ranges or when
+         * formatting a large number of selections.
+         *
+         * The given ranges are hints and providers can decide to format a smaller
+         * or larger range. Often this is done by adjusting the start and end
+         * of the range to full syntax nodes.
+         *
+         * @param document The document in which the command was invoked.
+         * @param ranges The ranges which should be formatted.
+         * @param options Options controlling formatting.
+         * @param token A cancellation token.
+         * @returns A set of text edits or a thenable that resolves to such. The lack of a result can be
+         * signaled by returning `undefined`, `null`, or an empty array.
```

Comment:
Since it is only stubbed, I don't think it really matters but `undefined` or `null` are not really supported by the signature, are they?

Reply:
It _would_ matter if we want to compile plugins against `theia.d.ts`, but `ProviderResult<T>` is defined as:

``` 
export type ProviderResult<T> = T | undefined | null | Thenable<T | undefined | null>;
```

Reply:
Ah, yes, I knew I was missing something obvious, thanks!